### ISSUE TITLE
Split assignments to separate controllers

### ISSUE THREAD

**Issue by NatTuck** [CodeGrade/bottlenose#23]
Currently we have three assignment types:
- Files
- Questions
- Exams

We should split these to separate controllers so they can have separate views.

That probably implies separate models and single-table-inheritance and a subfolder under models.

This will also make it easier to add a couple new types:
 - Git
 - Just Points

This may require some more thought about type-specific graders, and the grader logic.

**Comment by blerner** [comment]
How do you go about creating separate subclasses of controllers?  We now have separate model subclasses, so that's most of the benefit already accomplished, I think...

**Comment by NatTuck** [comment]
Ah, here's where the plural thing matters.

You'd create a models_controller.rb for each model class, and then implement at least show methods in them. That'd let us split out the views, which would be useful.

**Comment by blerner** [comment]
Wouldn't that require changing routes also, so instead of /assignments/#, it'd be /files/# or /questions/# ? If so, I'm not in favor...

**Comment by NatTuck** [comment]
Yes, it would give us a /questions/# path. That's not a bad thing though, and the rails routing mechanics should handle it correctly. Much nicer than having to manually route one controller method to several views manually.

**Comment by blerner** [comment]
I disagree. It gives us 2n families of URLs, instead of two families (assignments, submissions, and their subtypes), of which most URLs don't work with most parameters. If I manually navigate to /files/123, but that assignment it actually a questions assignment, then it'll either crash accessing bogus fields (by trying to access files specific stuff), give nonsensical output (by actually giving questions specific stuff), or require a lot more error checking work on the part of each controller. As opposed to a single manual reflective dynamic dispatch in each controller.

There's no conceptual advantage to having those subtypes broken out into their own virtual collections, either: we're not going to have a /files index page separate from the /assignments index page, for instance.. 

It's not worth the trade-off. Rails doesn't have to dictate this structure to us if it doesn't improve our use cases in any material way.
